tachyon/zk/plonk/halo2/proof_reader.h

```diff
@@ -0,0 +1,268 @@
+#ifndef TACHYON_ZK_PLONK_HALO2_PROOF_READER_H_
+#define TACHYON_ZK_PLONK_HALO2_PROOF_READER_H_
+
+#include <utility>
+#include <vector>
+
+#include "tachyon/base/logging.h"
+#include "tachyon/crypto/transcripts/transcript.h"
+#include "tachyon/zk/plonk/halo2/proof.h"
+#include "tachyon/zk/plonk/keys/verifying_key.h"
+
+namespace tachyon::zk::halo2 {
+
+enum class ProofCursor {
```

Comment:
How about `ProofElement`?

Reply:
Since the `Reader` uses it, it's fine for the `Cursor` as well.

Reply:
I'll pick `ProofCursor` then.